components/compositing/constellation.rs

```diff
@@ -842,55 +845,56 @@ impl<LTF: LayoutThreadFactory, STF: ScriptThreadFactory> Constellation<LTF, STF>
     fn handle_send_error(&mut self, pipeline_id: PipelineId, err: IOError) {
         // Treat send error the same as receiving a panic message
         debug!("Pipeline {:?} send error ({}).", pipeline_id, err);
-        self.handle_panic(Some(pipeline_id), format!("Send failed ({})", err));
+        self.handle_panic(Some(pipeline_id), format!("Send failed ({})", err), String::from("<none>"));
     }
 
-    fn handle_panic(&mut self, pipeline_id: Option<PipelineId>, reason: String) {
+    fn handle_panic(&mut self, pipeline_id: Option<PipelineId>, reason: String, backtrace: String) {
+        error!("Panic: {}", reason);
+        if !self.handled_panic || opts::get().full_backtraces {
```

Comment:
Perhaps we should print that a backtrace was skipped otherwise? (Perhaps only once)

I'm a bit concerned that this may lead to a situation where a panic cascade panic reaches the constellation first (not sure if that is possible), leading to a confusing backtrace. Of course, `-Z full-backtraces` is still an option, but the user should know that backtraces have been magicked away to decide to use it :smile: